### PR TITLE
Burn Issue Fix

### PR DESCRIPTION
**Pull request by ElliotFriedman**

### CODE REVIEW

**Review by juniuszhou — APPROVED** [review]

**Review by utx0 — APPROVED** [review]
Nice work

**Comment by ElliotFriedman** [comment]
This will need to be reworked. Whitelists should only include one type of token. Either cosmos native or ethereum native, but not both.

File: smart-contracts/contracts/BridgeBank/BridgeBank.sol [hunk 1]
```diff
@@ -197,7 +231,7 @@ contract BridgeBank is BankStorage, CosmosBank, EthereumBank, WhiteList {
         bytes memory _recipient,
         address _token,
         uint256 _amount
-    ) public payable onlyWhiteList(_token) {
+    ) public payable onlyEthTokenWhiteList(_token) validSifAddress(_recipient) {
         string memory symbol;
```
Comment (ElliotFriedman) [inline smart-contracts/contracts/BridgeBank/BridgeBank.sol:235]: There is an ethereum token whitelist and a cosmos token whitelist. This keeps the concerns separate and avoids attack vectors where a user could try to burn a whitelisted token

File: smart-contracts/contracts/BridgeBank/CosmosWhiteList.sol [hunk 2]
```diff
@@ -17,16 +17,16 @@ contract WhiteList is WhiteListStorage {
...
 
     /*
      * @dev: Modifier to restrict erc20 can be locked
      */
-    modifier onlyWhiteList(address _token) {
+    modifier onlyCosmosTokenWhiteList(address _token) {
         require(
-            getTokenInWhiteList(_token),
+            getCosmosTokenInWhiteList(_token),
             "Only token in whitelist can be transferred to cosmos"
```
Comment (stbeyer) [inline smart-contracts/contracts/BridgeBank/CosmosWhiteList.sol:30]: The error message should probably change

**Review by stbeyer — APPROVED** [review]
Looks good. Left a comment about error string, since they should be different on the two whitelists, I suppose

**Review by utx0 — APPROVED** [review]
All look good to me mate. Merging now.

**Pull request merged**